orchagent/portsorch.cpp

```diff
@@ -174,11 +177,45 @@ bool PortsOrch::setPortAdminStatus(sai_object_id_t id, bool up)
 
     sai_status_t status = sai_port_api->set_port_attribute(id, &attr);
     if (status != SAI_STATUS_SUCCESS)
-    {
         return false;
+    return true;
+}
+
+bool PortsOrch::setHostIntfsOperStatus(sai_object_id_t port_id, bool up)
+{
+    SWSS_LOG_ENTER();
+
+    for (auto it = m_portList.begin(); it != m_portList.end(); it++)
+    {
+        if (it->second.m_port_id == port_id)
+        {
+            sai_attribute_t attr;
+            attr.id = SAI_HOSTIF_ATTR_OPER_STATUS;
+            attr.value.booldata = up;
+
+            sai_status_t status = sai_hostif_api->set_hostif_attribute(it->second.m_hif_id, &attr);
+            if (status != SAI_STATUS_SUCCESS)
+                return false;
+            return true;
+        }
     }
+    return false;
+}
 
-    return true;
+void PortsOrch::updateDbPortOperStatus(sai_object_id_t id, sai_port_oper_status_t status)
+{
+    SWSS_LOG_ENTER();
+
+    for (auto it = m_portList.begin(); it != m_portList.end(); it++)
+    {
+        if (it->second.m_port_id == id)
+        {
+            vector<FieldValueTuple> vector;
+            FieldValueTuple tuple("oper_status", to_string(status));
+            vector.push_back(tuple);
+            m_portTable->set(it->first, vector);
```

Comment:
currently no. it will be supported by the modifications in swss-common library, which I am currently working on.